code/game/objects/items/stacks/stack.dm

```diff
@@ -441,13 +441,15 @@
  * Arguments:
  * - [check][/obj/item/stack]: The stack to check for mergeability.
  */
-/obj/item/stack/proc/can_merge(obj/item/stack/check)
+/obj/item/stack/proc/can_merge(obj/item/stack/check, inhand = FALSE)
 	if(!istype(check, merge_type))
 		return FALSE
 	if(mats_per_unit ~! check.mats_per_unit) // ~! in case of lists this operator checks only keys, but not values
 		return FALSE
 	if(is_cyborg) // No merging cyborg stacks into other stacks
 		return FALSE
+	if(!(istype(check.loc, /obj/item/storage) || isturf(check.loc)) && !inhand) // no merging with items that arent on the ground or in the backpack (aka, item slots)
+		return FALSE
```

Comment:
do we really want to relegate being "in storage" to only these subtypes?
what about ore chests, etc etc.
Maybe it would be better to simply do `if(!ismob)`